pkg/backends/newrelic/newrelic.go

```diff
@@ -88,14 +90,22 @@ type Client struct {
 	flushInterval    time.Duration
 }
 
-// NewRelicInfraPayload struct
-type NewRelicInfraPayload struct {
+// NRInfraPayload represents New Relic Infrastructure Payload format
+// https://github.com/newrelic/infra-integrations-sdk/blob/master/docs/v2tov3.md#v2-json-full-sample
+type NRInfraPayload struct {
 	Name               string        `json:"name"`
 	ProtocolVersion    string        `json:"protocol_version"`
 	IntegrationVersion string        `json:"integration_version"`
 	Data               []interface{} `json:"data"`
 }
 
+// NRMetricsPayload represents New Relic Metrics Payload format
+// https://docs.newrelic.com/docs/data-ingest-apis/get-data-new-relic/metric-api/report-metrics-metric-api#new-relic-guidelines
+type NRMetricsPayload struct {
+	Common  map[string]interface{} `json:"common"`
```

Comment:
Can `Common` and `Metrics` be more well defined rather than a `map[string]interface{}`?

In the `NRInfraPayload` it uses metric names as keys, so it can't be well defined, but in `NRMetricsPayload` it has an actual `name` key, so it can.

Ideally everything that can possibly be well defined should be.

Reply:
I've added both a common and metric struct now. 
NRMetric struct is used within flush.go, I have intentionally left the NRMetricsPayload struct as []interface{} rather then []NRMetric so that the timeseries struct in flush.go can remain common across the flush types used.